libs/output-mapping/src/DeferredTasks/FailedLoadTableDecider.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Keboola\OutputMapping\DeferredTasks;
+
+use Keboola\StorageApi\Client;
+use Keboola\StorageApi\ClientException;
+use Psr\Log\LoggerInterface;
+
+class FailedLoadTableDecider
+{
+    public static function decideTableDelete(
+        LoggerInterface $logger,
+        Client $client,
+        LoadTableTaskInterface $task
+    ): bool {
+        try {
+            $tableInfo = $client->getTable($task->getDestinationTableName());
+        } catch (ClientException $e) {
+            // likely the table doesn't exist, but any other error really prevents us from positive decision
+            return false;
+        }
+        if ($task->isUsingFreshlyCreatedTable() && // most important
+            ($tableInfo['rowsCount'] === 0 || $tableInfo['rowsCount'] === null) && // seems both are possible 🙄
+            (count($tableInfo['metadata']) === 0) // at this point there should be no metadata, they're set after load
```

Comment:
Zkontrolovat jestli podle metadat byla tabulka vytvořená touto konfigurací nejde, protože metadata se zapisují až po loadu, nicméně, to nahrazuje flag isUsingFreshlyCreatedTable kterej by měl být dost spolehlivej, to znamená, že tahle kontrola slouží vpodstatě jen na to, aby ověřila, že tabulku nezačal používat někdo jinej

Reply:
To je zralé na nějaký refactoring. Aby se část metadat nastavovala dřív a ne až po loadu.

Reply:
No to je k úvaze, ale tady by to stejně nijak nepomohlo, protože to by tam ty metadata byly zase vždycky :)